model/assembly/CompiledTestConverterService.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019  (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoDeliveryRdf\model\assembly;
+
+use InvalidArgumentException;
+use tao_models_classes_service_StorageDirectory;
+use oat\oatbox\filesystem\File;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoQtiTest\models\CompilationDataService;
+use oat\taoQtiTest\models\PhpCodeCompilationDataService;
+use oat\taoQtiTest\models\XmlCompilationDataService;
+
+class CompiledTestConverterService extends ConfigurableService
+{
+    const SERVICE_ID = 'taoDeliveryRdf/CompiledTestConverterService';
+
+    const OPTION_PHP_COMPILATION_SERVICE = 'php_compilation_service';
+    const OPTION_XML_COMPILATION_SERVICE = 'xml_compilation_service';
+
+    /**
+     * @var CompilationDataService
+     */
+    private $phpCompilationService = null;
+
+    /**
+     * @var CompilationDataService
+     */
+    private $xmlCompilationService = null;
+
+    /**
+     * CompiledTestConverterService constructor.
+     * @param array $options
+     */
+    public function __construct($options = array())
+    {
+        parent::__construct($options);
+
+        $this->phpCompilationService = $this->getOption(self::OPTION_PHP_COMPILATION_SERVICE);
+        if (!$this->phpCompilationService instanceof PhpCodeCompilationDataService) {
+            throw new InvalidArgumentException(sprintf('%s option must be an instance of %s', self::OPTION_PHP_COMPILATION_SERVICE,  PhpCodeCompilationDataService::class));
+        }
+
+        $this->xmlCompilationService = $this->getOption(self::OPTION_XML_COMPILATION_SERVICE);
+        if (!$this->xmlCompilationService instanceof XmlCompilationDataService) {
+            throw new InvalidArgumentException(sprintf('%s option must be an instance of %s', self::OPTION_XML_COMPILATION_SERVICE, XmlCompilationDataService::class));
+        }
+    }
+
+    /**
+     * @param File                                          $file
+     * @param tao_models_classes_service_StorageDirectory   $directory
+     * @return File
+     * @throws \common_Exception
+     */
+    public function convertPhpToXml(File $file, tao_models_classes_service_StorageDirectory $directory)

```

Comment:
you can combine converts methods to make it in 1 method, they do the same things

Reply:
I agree that we can extract some common logic into separate method. But in general when there are two different methods it's clear what you have to use from client code, based on your needs.